### ISSUE TITLE
Changelog always generated twice

### ISSUE THREAD

**Issue by nfantone** [conventional-changelog/conventional-changelog#540]
I'm trying out the tool in a new repository and I'm always getting the entire changelog generated and appended to the existing file, regardless what CLI options I use.

Here's what I'm doing:

1) Clone fresh repository (no `CHANGELOG.md` file on it).
2) Generate entire changelog.

```sh
npx -p conventional-changelog-cli conventional-changelog -p angular -i CHANGELOG.md -s -t '' -r 0
```

3) `CHANGELOG.md` gets generated ok.

```md
## 0.1.20 (2019-10-25)


### Bug Fixes

...

### Features

...

```

4) Bump version in `package.json`.
5) Re-run changelog generation without `-r 0` flag option.

```sh
npx -p conventional-changelog-cli conventional-changelog -p angular -i CHANGELOG.md -s -t ''
```

6) New version header gets added at the top as expected. However, **whole changelog is appended to the bottom** of the previously generated log.

```md
## 0.1.21 (2019-10-29) -> NEW VERSION


## 0.1.20 (2019-10-25)

...

# 0.1.0 (2019-09-30) -> OLDEST VERSION

...

 
## 0.1.21 (2019-10-29) -> NEW VERSION AGAIN (WHOLE CHANGELOG DUPLICATED BELOW)

...

```

What am I doing wrong here?

**Comment by nfantone** [comment]
Tried the above with `2.0.25` and `next` versions (`2.0.16`). Both reveal the same behaviour.

As a workaround, I'm always generating the entire changelog file on each release (`-r 0`) which produces the expected result each time.

**Comment by tommywo** [comment]
Confirming there is a bug - I could kinda reproduce it, but with different results.
It doesn't happen without `-t ''` flag, so I think the cause is that tagPrefix is set to default `v`, last tag cannot be found, so it would generate whole changelog again including previous versions, and put on  the top of changelog.

**Comment by nfantone** [comment]
@tommywo Interesting. Unfortunately, in my scenario, this is an old project with several existing tagged versions - and they do not have any prefix (i.e.: `0.2.0`).

Any thoughts on why this happens? Why would it _not_ found the last tag according to your explanation?

**Comment by tommywo** [comment]
@nfantone apperantly runnining `conventional-changelog -p angular -i CHANGELOG.md -s` without `-t ''` works with tags like `0.1.0`

**Comment by tommywo** [comment]
Strange thing with `meow` is that with tagPrefix setup like this:
```
'tag-prefix': {
      alias: 't'    
    }
```
`-t ''` will parse to `tagPrefix: true`, and `--issue-prefix ''` to ` tagPrefix: ''`
and with below setup like
```
'tag-prefix': {
      alias: 't',
      type: 'string'
    }
```
both will parse to  ` tagPrefix: ''`

**Comment by tommywo** [comment]
fixed in conventional-changelog-cli@2.0.28
